### PR TITLE
Add datasets unittest

### PR DESCRIPTION
**Pull request by yinchimaoliang**
Add datasets unittest, change data_augment_utils.py.

### CODE REVIEW

File: mmdet3d/datasets/pipelines/data_augment_utils.py [hunk 1]
```diff
@@ -63,19 +63,15 @@ def box_collision_test(boxes, qboxes, clockwise=True):
...
                             C = lines_qboxes[j, box_l, 0]
                             D = lines_qboxes[j, box_l, 1]
-                            acd = (D[1] - A[1]) * (C[0] -
-                                                   A[0]) > (C[1] - A[1]) * (
-                                                       D[0] - A[0])
-                            bcd = (D[1] - B[1]) * (C[0] -
-                                                   B[0]) > (C[1] - B[1]) * (
-                                                       D[0] - B[0])
+                            acd = (D[1] - A[1]) * (C[0] - A[0]) > (
+                                C[1] - A[1]) * (D[0] - A[0]) + 1e-5
```
Comment (ZwwWayne) [inline mmdet3d/datasets/pipelines/data_augment_utils.py:67]: Why do we need `1e-5` here?

File: tests/test_dataset/test_lyft_dataset.py [hunk 2]
```diff
@@ -76,38 +76,37 @@ def test_getitem():
...
+    self = LyftDataset(ann_file, None, root_path, classes=tmp_file.name)
+    assert self.CLASSES != original_classes
+    assert self.CLASSES == ['car', 'pedestrian']
 
 
 def test_evaluate():
     root_path = './tests/data/lyft'
     ann_file = './tests/data/lyft/lyft_infos_val.pkl'
-    lyft_dataset = LyftDataset(ann_file, None, root_path)
+    self = LyftDataset(ann_file, None, root_path)
```
Comment (ZwwWayne) [inline tests/test_dataset/test_lyft_dataset.py:108]: Change lyft_dataset to self is unnecessary.

File: tests/test_dataset/test_scannet_dataset.py [hunk 3]
```diff
@@ -89,26 +89,26 @@ def test_getitem():
...
         root_path, ann_file, pipeline=None, classes=['cabinet', 'bed'])
-    assert scannet_dataset.CLASSES != original_classes
-    assert scannet_dataset.CLASSES == ['cabinet', 'bed']
+    assert self.CLASSES != original_classes
+    assert self.CLASSES == ['cabinet', 'bed']
 
-    scannet_dataset = ScanNetDataset(
+    self = ScanNetDataset(
         root_path, ann_file, pipeline=None, classes=('cabinet', 'bed'))
-    assert scannet_dataset.CLASSES != original_classes
```
Comment (ZwwWayne) [inline tests/test_dataset/test_scannet_dataset.py:99]: We do not need to change `scannet_dataset` to `self`.

File: tests/test_pipeline/test_data_augment_utils.py [hunk 4]
```diff
@@ -23,7 +23,7 @@ def test_noise_per_object_v3_():
     expected_gt_bboxes_3d = np.array(
         [[3.3430212, 2.1475432, 9.388738, 1.2, 1.89, 0.48, 0.05056486]])
 
-    assert points.shape == (800, 4)
+    assert points.shape == (1000, 4)
```
Comment (ZwwWayne) [inline tests/test_pipeline/test_data_augment_utils.py:26]: It is weird to change `800` to` 1000`.

File: mmdet3d/datasets/pipelines/data_augment_utils.py (hunk 1 above)
Comment (yinchimaoliang) [reply mmdet3d/datasets/pipelines/data_augment_utils.py:67]: In different environments, the input of this function is different, resulting in different output. by adding 1e-5, we can get the same output in different environments.

File: tests/test_dataset/test_kitti_dataset.py [hunk 5]
```diff
@@ -0,0 +1,370 @@
...
+        torch.tensor(
+            [[8.7314, -1.8559, -1.5997, 0.4800, 1.2000, 1.8900, 0.0100]]))
+    labels_3d = torch.tensor([
+        0,
+    ])
+    scores_3d = torch.tensor([0.5])
+    metric = ['mAP']
+    result = dict(boxes_3d=boxes_3d, labels_3d=labels_3d, scores_3d=scores_3d)
+    ap_dict = self.evaluate([result], metric)
+    assert abs(ap_dict['KITTI/Overall_3D_easy'] - 3.0303030303030307) < 1e-5
```
Comment (ZwwWayne) [inline tests/test_dataset/test_kitti_dataset.py:153]: Try use np.allclose or torch.allclose.

**Pull request merged**